File: daybreak/src/daybreak/adapters/terminal/builders.py
from daybreak.terminals.config_terminals import Ghostty, WezTerm
from daybreak.terminals.kitty import Kitty
from daybreak.terminals.konsole import Konsole
from daybreak.terminals.neovim import Neovim

from .windows_terminal import WindowsTerminalAdapter
from .wrappers import LegacyModeTerminalAdapter, UniversalPtyAdapter


def build_linux_terminal_adapters():
    return [
        UniversalPtyAdapter(),
        LegacyModeTerminalAdapter(Neovim()),
        LegacyModeTerminalAdapter(Kitty()),
        LegacyModeTerminalAdapter(Ghostty()),
        LegacyModeTerminalAdapter(WezTerm()),
        LegacyModeTerminalAdapter(Konsole()),
    ]


def build_windows_terminal_adapters():
    return [
        WindowsTerminalAdapter(),
    ]

File: src/adapters/terminal/builders.ts
import { Ghostty, WezTerm } from '../../terminals/configTerminals';
import { Kitty } from '../../terminals/kitty';
import { Konsole } from '../../terminals/konsole';
import { Neovim } from '../../terminals/neovim';
import { WindowsTerminalAdapter } from './windowsTerminal';
import { LegacyModeTerminalAdapter, UniversalPtyAdapter } from './wrappers';

export function buildLinuxTerminalAdapters() {
  return [
    new UniversalPtyAdapter(),
    new LegacyModeTerminalAdapter(new Neovim()),
    new LegacyModeTerminalAdapter(new Kitty()),
    new LegacyModeTerminalAdapter(new Ghostty()),
    new LegacyModeTerminalAdapter(new WezTerm()),
    new LegacyModeTerminalAdapter(new Konsole()),
  ];
}

export function buildWindowsTerminalAdapters() {
  return [
    new WindowsTerminalAdapter(),
  ];
}
